### PR TITLE
[master][base_vat] Filter option by partner with vat number added.

### PR DESCRIPTION
**Pull request by sabrinagnu**
In this branch we add the filter option by partner with vat number.

### CODE REVIEW

**Comment by odony** [comment]
As @beledouxdenis explained on the original PR #1389, this change has too many global consequences to be merged.
If you want to simply add the VAT field in the search view of partner, that's an option. And if you want to allow name_search() to match for a VAT number (but *only* a VAT number, not some pattern), that's a possibility too, in master. But everything else is too much, especially adding the VAT number everywhere in name_get all the time - it will annoy everybody, will be redundant in most places where VAT is already displayed, etc.

In other words, you could redo the PR with only the following if you want a better chance to get it merged ;-)
 - add the vat field to the partner search view
 - inherit `name_search` to perform an *exact* search on the vat number before calling `super()`, in case the user copy/pasted a vat number. No pattern matching or anything else. And you should only do it when the `name_search` `operator` is a *positive* operator. See [this other one](https://github.com/odoo/odoo/blob/99b7a6e7a16/addons/product/product.py#L1104-L1107) for example.

Thanks!
